protocol/message.go

```diff
@@ -93,3 +127,55 @@ func (m *Message) Decrypt(cipherStr string, keys map[string]string) error {
 
 	return nil
 }
+
+func (m *Message) Encrypt(cipherStr string, keys map[string]string) error {
```

Comment:
`keys map[string]string` probably need to just be `key string`. I thought I would need the `iv` at some point but it is all random.